File: bindings/csharp/Satox/Core/SatoxError.cs
using System;

namespace Satox.Core
{
    /// <summary>
    /// SatoxError represents an error that occurred in the Satox SDK.
    /// </summary>
    public class SatoxError : Exception
    {
        /// <summary>
        /// Creates a new SatoxError with the specified message.
        /// </summary>
        /// <param name="message">The error message.</param>
        public SatoxError(string message) : base(message)
        {
        }

        /// <summary>
        /// Creates a new SatoxError with the specified message and cause.
        /// </summary>
        /// <param name="message">The error message.</param>
        /// <param name="innerException">The cause of the error.</param>
        public SatoxError(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
} 

File: bindings/java/src/main/java/com/satox/core/SatoxError.java
package com.satox.core;

public class SatoxError extends Exception {
    private String code;

    public SatoxError(String message) {
        super(message);
        this.code = "UNKNOWN_ERROR";
    }

    public SatoxError(String message, String code) {
        super(message);
        this.code = code;
    }

    public String getCode() { return code; }
    public void setCode(String code) { this.code = code; }
} 